File: 프로그래머스/2/131701. 연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.py
def solution(elements):
    comb = set()
    
    elLen = len(elements)
    elements = elements * 2
    
    for i in range(elLen):
        for j in range(elLen):
            comb.add(sum(elements[j:j+i+1]))
    
    return len(comb)

File: 프로그래머스/2/131701. 연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.js
function solution(elements) {
    const sum = (arr) => arr.reduce((a, c) => a + c, 0);
    
    let comb = new Set();
    const element = elements.concat(elements);
    
    for (let i = 0; i < elements.length; i++){
        for (let j = 0; j < elements.length; j++){
            comb.add(sum(element.slice(i, i+j+1)))
        }
    }
    
    return comb.size;
}